### PR TITLE
DDF-4386: Missing metacard definition handling and feedback

### PR DESCRIPTION
**Pull request by leo-sakh**
#### What does this PR do?
Updates the manual metacard builder so that it can handle missing metacard definitions.

#### Who is reviewing it? 
@Schachte 
@glenhein 
@brianfelix 
@bellcc 
#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler
@bdeining

#### How should this be tested?
1. Make sure the `Metacard Type Whitelist` is empty in admin console
2. Check that manual metacard builder is working
3. Add a filter to `Metacard Type Whitelist` in admin console
4. Check that manual opening the manual metacard builder displays a warning message

#### What are the relevant tickets?
[DDF-4386](https://codice.atlassian.net/browse/DDF-4386)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/component/property/property.collection.view.js [hunk 1]
```diff
@@ -389,6 +399,7 @@ module.exports = Marionette.CollectionView.extend(
           return false
         }
       })
+
```
Comment (bellcc) [inline ui/packages/catalog-ui-search/src/main/webapp/component/property/property.collection.view.js]: ✏️ Remove this newline

File: ui/packages/catalog-ui-search/src/main/webapp/component/builder/builder.view.js [hunk 2]
```diff
@@ -35,6 +34,13 @@ const ajaxCall = $.get({
   availableTypes = response
 })
 
+const announceMissingDefinition = definition =>
+  announcement.announce({
+    title: 'Missing Metacard Definition',
```
Comment (Bdthomson) [inline ui/packages/catalog-ui-search/src/main/webapp/component/builder/builder.view.js]: Do we use `metacard` anywhere in the ui? I thought that was an internal implementation detail.

File: ui/packages/catalog-ui-search/src/main/webapp/component/property/property.collection.view.js [hunk 3]
```diff
@@ -352,22 +358,26 @@ module.exports = Marionette.CollectionView.extend(
...
       }, [])
-      var typeIntersection = _.intersection.apply(
+      const typeIntersection = _.intersection.apply(
         _,
-        metacardTypes.map(type => {
-          return Object.keys(metacardDefinitions.metacardDefinitions[type])
-        })
+        metacardTypes
+          .filter(
+            type => metacardDefinitions.metacardDefinitions[type] !== undefined
```
Comment (andrewkfiedler) [inline ui/packages/catalog-ui-search/src/main/webapp/component/property/property.collection.view.js]: ❓ This is to fix issues with missing types due to their lack of whitelisting right?  What does it look like when looking at a metacard with a missing type?  Do we still show things like title?

File: ui/packages/catalog-ui-search/src/main/webapp/component/property/property.collection.view.js [hunk 4]
```diff
@@ -287,22 +287,28 @@ module.exports = Marionette.CollectionView.extend(
...
-          required: false,
-        })
-      })
+      let propertyIntersection = this.determinePropertyIntersection(metacards)
+
+      if (
+        Array.isArray(propertyIntersection) &&
+        propertyIntersection.length === 0
+      ) {
+        propertyIntersection = metacards.flatMap(card => Object.keys(card))
```
Comment (andrewkfiedler) [inline ui/packages/catalog-ui-search/src/main/webapp/component/property/property.collection.view.js]: ❓ Is this a workaround for missing metacard types?

File: ui/packages/catalog-ui-search/src/main/webapp/component/property/property.collection.view.js (hunk 3 above)
Comment (leo-sakh) [reply ui/packages/catalog-ui-search/src/main/webapp/component/property/property.collection.view.js]: If we don't have the type we can't find various field definitions.

File: ui/packages/catalog-ui-search/src/main/webapp/component/property/property.collection.view.js (hunk 4 above)
Comment (leo-sakh) [reply ui/packages/catalog-ui-search/src/main/webapp/component/property/property.collection.view.js]: Yeah, I forgot to remove this during my various stabs at fixing it. I removed it.

File: ui/packages/catalog-ui-search/src/main/webapp/component/builder/builder.view.js (hunk 2 above)
Comment (leo-sakh) [reply ui/packages/catalog-ui-search/src/main/webapp/component/builder/builder.view.js]: Talked with @millerw8 and changed the wording.

File: ui/packages/catalog-ui-search/src/main/webapp/component/property/property.collection.view.js (hunk 1 above)
Comment (leo-sakh) [reply ui/packages/catalog-ui-search/src/main/webapp/component/property/property.collection.view.js]: Removed

File: ui/packages/catalog-ui-search/src/main/webapp/component/builder/builder.view.js [hunk 5]
```diff
@@ -146,52 +150,64 @@ module.exports = Marionette.LayoutView.extend({
...
+    if (!selection) {
+      announceMissingDefinition(selectedAvailableType)
+      return
+    }
     this.model.set(
       'selectedAvailableType',
-      availableTypes.filter(
-        availableType => availableType.metacardType === selectedAvailableType
-      )[0]
+      selected
```
[the remark this answers is not in the document]
Comment (leo-sakh) [reply ui/packages/catalog-ui-search/src/main/webapp/component/builder/builder.view.js]: 👍 good catch!

File: ui/packages/catalog-ui-search/src/main/webapp/component/builder/builder.less [hunk 6]
```diff
@@ -8,7 +8,7 @@
     width: 100%;
     overflow: auto;
     padding: 10px 10px;
-    max-height: ~'calc(100% - 1*@{minimumButtonSize} - 10px)';
+    max-height: ~'calc(100% - 3.5*@{minimumButtonSize} - 10px)';
```
Comment (leo-sakh) [inline ui/packages/catalog-ui-search/src/main/webapp/component/builder/builder.less]: Before, using `1*@{min...}`, the footer would be invisible to the users

**Comment by bellcc** [comment]
### Hero Succesful 🎉 

**Steps**
- Verified that whitelist is empty by default
- Created list with a nonsensical metacard type
- Verified that popup does come up

**Full Build Complete**
```
In progress
```

**Pull request merged**